src/test/fuzz.cpp

```diff
@@ -147,84 +62,64 @@ fuzz(std::string const& filename, el::Level logLevel,
     while (__AFL_LOOP(PERSIST_MAX))
 #endif // AFL_LLVM_MODE
     {
-        XDRInputFileStream in(MAX_MESSAGE_SIZE);
+        XDRInputFileStream in(XDR_SIZE_LIMIT);
         in.open(filename);
-        StellarMessage msg;
-        while (tryRead(in, msg))
-        {
-            // HELLO, AUTH and ERROR_MSG messages cause the connection between
-            // the peers to drop. Since peer connections are only established
-            // preceding the persistent loop, a dropped peer is not only
-            // inconvenient, it also confuses the fuzzer. Consider a msg A sent
-            // before a peer is dropped and after a peer is dropped. The two,
-            // even though the same message, will take drastically different
-            // execution paths -- the fuzzer's main metric for determinism
-            // (stability) and binary coverage.
-            if (msg.type() == HELLO || msg.type() == AUTH ||
-                msg.type() == ERROR_MSG)
-                continue;
-
-            LOG(INFO) << "Fuzzer injecting message." << msgSummary(msg);
-
-            seedRandomness();
-
-            auto nodeids = simulation->getNodeIDs();
-
-            auto loopbackPeerConnection = simulation->getLoopbackConnection(
-                nodeids[INITIATOR], nodeids[ACCEPTOR]);
 
-            // ensure connection exists
-            assert(loopbackPeerConnection);
-
-            auto initiator = loopbackPeerConnection->getInitiator();
-            auto acceptor = loopbackPeerConnection->getAcceptor();
-
-            initiator->getApp().getClock().postToCurrentCrank(
-                [initiator, msg]() { initiator->Peer::sendMessage(msg); });
-
-            simulation->crankForAtMost(std::chrono::milliseconds{500}, false);
-
-            // clear all queues and cancel all events
-            initiator->clearInAndOutQueues();
-            acceptor->clearInAndOutQueues();
-
-            while (initiator->getApp().getClock().cancelAllEvents())
-                ;
-            while (acceptor->getApp().getClock().cancelAllEvents())
-                ;
-        }
+        fuzzer->inject(in);
     }
 }
 
 void
-genfuzz(std::string const& filename)
+genfuzz(std::string const& filename, FuzzerMode fuzzerMode)
```

Comment:
Not quite sure I follow here. 

Currently each fuzzer has a corresponding free function in the `FuzzerImpl` file with generation logic, that shares `isBadInput` with the fuzzer. These generation methods must either be free or static, because we do not need (and do not want) to initialize a fuzzer and start up apps just to create a random XDR. That being said, I could separate the initialization of the fuzzer to a separate method, and out of the constructor but I am not sure if this is the best route.

Reply:
yes, that's what I would like to see (switch on `mode` is basically what polymorphism does, in a cleaner way):
that way, each fuzzing mode is encapsulated in a dedicated class, if you move what is currently in the constructor into a method you'll have what you want.

Reply:
This should be addressed in my latest commits.